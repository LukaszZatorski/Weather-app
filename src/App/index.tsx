import React, { useState, useEffect } from 'react';
import './index.scss';
import CurrentWeather from '../CurrentWeather';
import LocationSearch from '../LocationSearch';

type LatLng = {
  lat: number;
  lng: number;
};

const GOOGLE_API_KEY = process.env.REACT_APP_GOOGLE_MAP_API_KEY;

const App = () => {
  const [latLng, setLatLng] = useState();

  useEffect(() => {
    const script = document.createElement('script');

    script.src = `https://maps.googleapis.com/maps/api/js?key=${GOOGLE_API_KEY}&libraries=places&callback=placesAutocompleteCallback`;
    script.async = true;

    document.body.appendChild(script);
  }, []);

  const updateLatLng = (latLng: LatLng) => {
    setLatLng(latLng);
  };
  return (
    <div className="App">
      <header className="App-header">
        <h4>SimpleWeather</h4>
        <LocationSearch updateLatLng={updateLatLng} />
      </header>
      <CurrentWeather latLng={latLng} />
    </div>
  );
};

export default App;
