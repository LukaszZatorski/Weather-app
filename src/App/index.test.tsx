import React from 'react';
import { render } from '@testing-library/react';
import App from '.';

describe('App', () => {
  test('renders without crashing', () => {
    render(<App />);
  });
  test('has a valid snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
