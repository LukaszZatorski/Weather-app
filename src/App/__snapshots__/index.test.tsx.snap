// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h4>
        SimpleWeather
      </h4>
      <div
        class="Location-search"
      >
        <div
          class="Search-box"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            autocomplete="off"
            class="location-search-input"
            disabled=""
            placeholder="Enter City..."
            role="combobox"
            type="text"
            value=""
          />
        </div>
        <span
          class="Geolocation"
        >
          <i
            class="fas fa-map-marker-alt fa-2x"
          />
        </span>
      </div>
    </header>
    <div
      class="Current-weather"
    />
    <div
      class="Forecast"
    />
  </div>
</div>
`;
